### PR TITLE
Indicate to the user that a stash save is in progress

### LINKED ISSUE #1495: Add a visual cue to indicate that stashing is in progress
**Is your feature request related to a problem? Please describe.**
When stashing it can take several seconds before the stash completes and the popup closes, but there is no visual cue that it is happening. I have on occasion wondered whether I actually pressed Enter and have pressed it again; fortunately, this seems to cause no harm.

**Describe the solution you'd like**
In the Stash popup, overwrite the "type name (optional)" message or the name (if typed) with a message like "Stashing..." as soon as Enter is pressed. This will inform the user that something is actually happening.

### PR DESCRIPTION
**Pull request by nrempel**
This Pull Request fixes #1495.

It changes the following:
- Add AsyncStash
- Updates `StashMsgComponent` to use AsyncStash instead of sync::Stash
- Introduces disabled flag on `TextInputComponent`
- Change `StashMsgComponent` title to "Stashing..." while stash is pending and disable input

I followed the checklist:
- [ ] I added unittests
- [x] I ran `make check` without errors
- [x] I tested the overall application
- [x] I added an appropriate item to the changelog

### CODE REVIEW

File: src/components/textinput.rs [hunk 1]
```diff
@@ -31,11 +31,13 @@ pub enum InputType {
 }
 
 /// primarily a subcomponet for user input of text (used in `CommitComponent`)
+#[allow(clippy::struct_excessive_bools)]
```
Comment (nrempel) [inline src/components/textinput.rs:34]: Note: I saw this elsewhere so I also allowed this rule. Refactoring this not to include 3 bools didn't make sense to me.

File: src/components/textinput.rs [hunk 2]
```diff
@@ -419,7 +432,7 @@ impl Component for TextInputComponent {
 	}
 
 	fn event(&mut self, ev: &Event) -> Result<EventState> {
-		if self.visible {
+		if self.visible && !self.disabled {
```
Comment (extrawurst) [inline src/components/textinput.rs:435]: this is pretty invasive. why do we need that? the text input for the stashing can just close IMO and does not need to learn a new state of `disabled`. what is the rational?

**Review by extrawurst — CHANGES_REQUESTED** [review]

**Comment by xcloudplatform** [comment]
yes we need that

File: src/components/textinput.rs (hunk 2 above)
Comment (extrawurst) [reply src/components/textinput.rs:435]: @nrempel

Comment (nrempel) [reply src/components/textinput.rs:435]: Without disabled, it's possible to type in the field while the operation is pending.

I suppose we could close the model and continue the operation in the background. In this case, is there a pattern for handling stashing errors if they occur?

Comment (extrawurst) [reply src/components/textinput.rs:435]: yeah that's what I think we should do, what we do in other cases already. just close the text input popup regularly and then the new popup will block any other input and visualise that the stashing is in progress

**Comment by extrawurst** [comment]
@nrempel are you willing to finish this up?

**Comment by nrempel** [comment]
Sorry, it's been a very busy year. I may have time to finish this over the holiday break. But I'll close this for now and revive it if needed.

**Pull request closed without merging**